AzureMonitorAgent/agent.py

```diff
@@ -588,54 +484,197 @@ def enable():
+            default_configs["MONITORING_GCS_CERT_CERTFILE"] = "/etc/opt/microsoft/azuremonitoragent/gcscert.pem"
+            with open("/etc/opt/microsoft/azuremonitoragent/gcscert.pem", "wb") as f:
+                f.write(MONITORING_GCS_CERT_CERTFILE)
+            os.chown("/etc/opt/microsoft/azuremonitoragent/gcscert.pem", uid, gid)
+            os.system('chmod {1} {0}'.format("/etc/opt/microsoft/azuremonitoragent/gcscert.pem", 400))
+
+        if MONITORING_GCS_CERT_KEYFILE is not None:
+            default_configs["MONITORING_GCS_CERT_KEYFILE"] = "/etc/opt/microsoft/azuremonitoragent/gcskey.pem"
+            with open("/etc/opt/microsoft/azuremonitoragent/gcskey.pem", "wb") as f:
+                f.write(MONITORING_GCS_CERT_KEYFILE)
+            os.chown("/etc/opt/microsoft/azuremonitoragent/gcskey.pem", uid, gid)
+            os.system('chmod {1} {0}'.format("/etc/opt/microsoft/azuremonitoragent/gcskey.pem", 400))
+
+        if MONITORING_TENANT != "":
+            default_configs["MONITORING_TENANT"] = MONITORING_TENANT
+
+        if MONITORING_ROLE != "":
+            default_configs["MONITORING_ROLE"] = MONITORING_ROLE
+
+        if MONITORING_TENANT != "":
+            default_configs["MONITORING_ROLE_INSTANCE"] = MONITORING_ROLE_INSTANCE
+
+def handle_mcs_config(public_settings, protected_settings, default_configs):
+    """
+    Populate the defaults for MCS mode
+    """
+    default_configs["ENABLE_MCS"] = "true"
+    default_configs["PA_GIG_BRIDGE_MODE"] = "true"
+    # this port will be dynamic in future
+    default_configs["PA_FLUENT_SOCKET_PORT"] = "13000"
+
+    # fetch proxy settings
+    if public_settings is not None and "proxy" in public_settings and "mode" in public_settings.get("proxy") and public_settings.get("proxy").get("mode") == "application":
+        default_configs["MDSD_PROXY_MODE"] = "application"
+
+        if "address" in public_settings.get("proxy"):
+            default_configs["MDSD_PROXY_ADDRESS"] = public_settings.get("proxy").get("address")
+        else:
+            log_and_exit("Enable", MissingorInvalidParameterErrorCode, 'Parameter "address" is required in proxy public setting')
+
+        if "auth" in public_settings.get("proxy") and public_settings.get("proxy").get("auth") == "true":
+            if protected_settings is not None and "proxy" in protected_settings and "username" in protected_settings.get("proxy") and "password" in protected_settings.get("proxy"):
+                default_configs["MDSD_PROXY_USERNAME"] = protected_settings.get("proxy").get("username")
+                default_configs["MDSD_PROXY_PASSWORD"] = protected_settings.get("proxy").get("password")
+            else:
+                log_and_exit("Enable", MissingorInvalidParameterErrorCode, 'Parameter "username" and "password" not in proxy protected setting')
+
+    # add managed identity settings if they were provided
+    identifier_name, identifier_value, error_msg = get_managed_identity()
```

Comment:
Nit: use managed_id_name, managed_id_value for better context compared to identifier_name, identifier_value